packages/sdk/src/__tests__/integration/modules/moduleFromEndpoints.spec.ts

```diff
@@ -19,12 +21,224 @@ describe("moduleFromEndpoints", () => {
 
   it("should use generic types to define the endpoints", async () => {
     const sdkConfig = {
-      commerce: buildModule(moduleFromEndpoints<Endpoints>),
+      commerce: buildModule(moduleFromEndpoints<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+      }),
     };
 
     const sdk = initSDK(sdkConfig);
 
     expect(sdk.commerce.getProduct).toBeDefined();
     expect(sdk.commerce.getProducts).toBeDefined();
   });
+
+  it("should allow to override the default HTTP Client", async () => {
+    const customHttpClient = jest
+      .fn()
+      .mockResolvedValue({ id: 1, name: "Test Product" });
+    const sdkConfig = {
+      commerce: buildModule(moduleFromEndpoints<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+        httpClient: customHttpClient,
+      }),
+    };
+    const sdk = initSDK(sdkConfig);
+
+    await sdk.commerce.getProduct({ id: 1 });
+
+    expect(customHttpClient).toHaveBeenCalled();
+  });
+
+  it("should send a POST request to <BASE_URL>/<METHOD_NAME> by default", async () => {
+    const customHttpClient = jest
+      .fn()
+      .mockResolvedValue({ id: 1, name: "Test Product" });
+    const sdkConfig = {
+      commerce: buildModule(moduleFromEndpoints<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+        httpClient: customHttpClient,
+      }),
+    };
+    const sdk = initSDK(sdkConfig);
+
+    await sdk.commerce.getProduct({ id: 1 });
+
+    expect(customHttpClient).toHaveBeenCalledWith(
+      "http://localhost:8181/commerce/getProduct",
+      {
+        method: "POST",
+        params: [{ id: 1 }],
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+      }
+    );
+  });
+
+  it("should use default HTTP Client if it's not provided", async () => {
+    const sdkConfig = {
+      commerce: buildModule(moduleFromEndpoints<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+      }),
+    };
+    const sdk = initSDK(sdkConfig);
+
+    const response = await sdk.commerce.getProduct({ id: 1 });
+
+    expect(response).toEqual({ id: 1, name: "Test Product" });
```

Comment:
TBH it would be great, but I cannot mock the defaultHTTPClient in here, even mocking on fetch would be rather testing the implementation so I'm running the actual call to the middleware and verifing the response.